src/main/cpp/src/shuffle_split_detail.hpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/utilities/traits.hpp>
+
+namespace spark_rapids_jni {
+
+namespace detail {
+
+/**
+ * @brief Functor for computing size of data elements for a given cudf type.
+ *
+ * Note: columns types which themselves inherently have no data (strings, lists,
+ * structs) return 0.
+ */
+struct size_of_helper {
+  template <typename T>
+  constexpr std::enable_if_t<!cudf::is_fixed_width<T>() && !std::is_same_v<T, cudf::string_view>,
+                             size_t>
+  operator()() const
+  {
+    return 0;
+  }
+
+  template <typename T>
+  constexpr std::enable_if_t<!cudf::is_fixed_width<T>() && std::is_same_v<T, cudf::string_view>,
+                             size_t>
+  operator()() const
+  {
+    return sizeof(cudf::device_storage_type_t<int8_t>);
+  }
+
+  template <typename T>
+  constexpr std::enable_if_t<cudf::is_fixed_width<T>(), size_t> __device__
+  operator()() const noexcept
+  {
+    return sizeof(cudf::device_storage_type_t<T>);
+  }
+};
+
+/**
+ * @brief Header for each partition.
+ */
+struct partition_header {
+  uint32_t magic_number;
+  uint32_t version;
+  uint64_t offset;
```

Comment:
In the java code this is a uint32_t not a unit64_t. Again I would prefer it if we could have the code match for both to start out with, and then we can make adjustments afterwards. I am fine if we have to throw an exception if the input does not fit.

https://github.com/NVIDIA/spark-rapids-jni/blob/74e33d948442d72cd45dee2f3e062603d4953284/src/main/java/com/nvidia/spark/rapids/jni/kudo/KudoTableHeader.java#L73

https://github.com/NVIDIA/spark-rapids-jni/blob/74e33d948442d72cd45dee2f3e062603d4953284/src/main/java/com/nvidia/spark/rapids/jni/kudo/KudoSerializer.java#L82